mono/metadata/sre.c

```diff
@@ -3533,7 +3533,9 @@ typebuilder_setup_fields (MonoClass *klass, MonoError *error)
 			def_values [i].data = (const char *)mono_image_alloc (image, len);
 			memcpy ((gpointer)def_values [i].data, p, len);
 		}
-		mono_dynamic_image_register_token (tb->module->dynamic_image, mono_metadata_make_token (MONO_TABLE_FIELD, first_idx + i), (MonoObject*)fb);
+
+		MonoObjectHandle field_builder_handle = MONO_HANDLE_NEW (MonoObject, fb);
```

Comment:
It's a little bit dangerous to do this outside of a `HANDLE_FUNCTION_ENTER/RETURN` pair - you are making a promise that all the callers of this method are themselves wrapped in enter/return or that the callers are icalls that have been converted to use coop handles.

In this case it looks like the only caller is `ves_icall_TypeBuilder_create_runtime_class` which has been changed to use handles.  So it's okay.